dragonfly/implementations/psyllid_provider.py

```diff
@@ -233,51 +235,181 @@ def get_number_of_streams(self, channel):
         return stream_count
 
 
-    # These methods shoud not be used yet
-    def set_fmt_snr_threshold(self, snr, channel='a'):
-        if self.mode_dict[channel] == 'streaming':
-            logger.warning('Psyllid not in streaming mode')
+    # trigger control
+    def set_trigger_configuration(self, threshold=16, threshold_high=0, n_triggers=1, channel='a'):
+        if self.mode_dict[channel] != 'triggered':
+            logger.warning('Psyllid not in triggered mode')
             return False
-        request = '.node-config.ch'+str(self.channel_dict[channel])+'.fmt.threshold-power-snr'
-        result = self.provider.set(self.queue_dict[channel]+request, snr)
-        self.deactivate(channel)
-        time.sleep(1)
-        self.activate(channel)
+
+        self._set_fmt_snr_threshold( threshold, channel)
+        if threshold_high > threshold:
+            self._set_fmt_snr_high_threshold( threshold_high, channel)
+            self._set_trigger_mode( "two-level-trigger", channel)
+        else:
+            self._set_trigger_mode( "single-level-trigger", channel)
+
+        result = self._set_n_triggers( n_triggers, channel)
+
+        if result != True:
+            return False
+
+        return True
+
+
+    def get_trigger_configuration(self, channel='a'):
+        if self.mode_dict[channel] != 'triggered':
+            logger.warning('Psyllid not in triggered mode')
+            return {'threshold': False, 'threshold_high' : False, 'n_triggers' : False}
+
+        threshold = self._get_fmt_snr_threshold( channel)
+        threshold_high = self._get_fmt_snr_high_threshold( channel)
+        n_triggers = self._get_n_triggers( channel)
+        trigger_mode = self._get_trigger_mode( channel )
+
+        # threshold_high !> threhold results in trigger_mode = 1
+        # threshold_high is not used in this case
+        if trigger_mode == 1:
+            threshold_high = None
+
+        return {'threshold': threshold, 'threshold_high' : threshold_high, 'n_triggers' : n_triggers}
+
+
+    # pre-trigger and skip-tolerance control
+    def set_time_window(self, pretrigger_time=2e-3, skip_tolerance=5e-3, channel='a'):
+        if self.mode_dict[channel] != 'triggered':
+            logger.warning('Psyllid not in triggered mode')
+            return False
+
+        self._set_pretrigger_time( pretrigger_time, channel)
```

Comment:
you're not checking the return value of `_set_pretrigger_time`; We should either be sure to check the return values of functions that return a boolean to indicate their success condition, or change the functions not to return anything if we don't think it is useful to check them.

(as an aside, I think I generally prefer not returning booleans but instead raising exceptions if something goes wrong) Again, this doesn't need to prevent us from merging the PR

Reply:
one of the returns was actually checked in set_trigger_configuration. The idea was that psyllid_provider returns True or False for everything (unless something unexpected happens) and the roach_daq_interface gets to decide whether the return should result in an error.
I currently don't see a reason why that couldn't be changed, so I will remove the boolean returns and raise errors in the psyllid_provider directly.